docs/02-app/01-building-your-application/02-data-fetching/03-patterns.mdx

```diff
@@ -310,3 +311,89 @@ The `utils/get-item` exports can be used by Layouts, Pages, or other components
 > **Good to know:**
 >
 > - We recommend using the [`server-only` package](/docs/app/building-your-application/rendering/composition-patterns#keeping-server-only-code-out-of-the-client-environment) to make sure server data fetching functions are never used on the client.
+
+## Tainting
```

Comment:
Since "taint" is not yet an established industry term, might consider expanding this header to be more descriptive, like "Preventing sensitive information from leaking to the client". Should also link to Seb's blog post which provides a more holistic explanation of how to think about security.